File: oo-patterns/src/main/java/victor/training/oo/creational/builder/Address.java
package victor.training.oo.creational.builder;

public class Address {
	private String streetName;
	private Integer streetNumber;
	private String city;
	private String country;

	public static class Builder {
		private final Address address = new Address();

		public Builder withStreetName(String streetName) {
			address.setStreetName(streetName);
			return this;
		}
		
		// SOLUTION(
		public Builder withStreetNumber(int streetNumber) {
			address.setStreetNumber(streetNumber);
			return this;
		}
		// SOLUTION)

		public Address build() {
			return address;
		}
	}

	public String getStreetName() {
		return streetName;
	}

	public void setStreetName(String streetName) {
		this.streetName = streetName;
	}

	public String getCity() {
		return city;
	}

	public void setCity(String city) {
		this.city = city;
	}

	public String getCountry() {
		return country;
	}

	public void setCountry(String country) {
		this.country = country;
	}

	public final Integer getStreetNumber() {
		return streetNumber;
	}

	public final void setStreetNumber(Integer streetNumber) {
		this.streetNumber = streetNumber;
	}

	
	
}

File: oo-patterns-php/victor/training/oo/creational/builder/Address.php
<?php
/**
 * Created by IntelliJ IDEA.
 * User: VictorRentea
 * Date: 9/18/2017
 * Time: 1:36 AM
 */

namespace training\oo\creational\builder;


class Address
{
    /* @var String */
    private $streetName;

    /* @var Integer */
    private $streetNumber;

    /* @var String */
    private $city;

    /* @var String */
    private $country;

    // INITIAL (
//    public function getStreetName(): String
//    {
//        return $this->streetName;
//    }
//
//    public function setStreetName(String $streetName): void
//    {
//        $this->streetName = $streetName;
//    }
//
//    public function getStreetNumber(): int
//    {
//        return $this->streetNumber;
//    }
//
//    public function setStreetNumber(int $streetNumber): void
//    {
//        $this->streetNumber = $streetNumber;
//    }
//
//    public function getCity(): String
//    {
//        return $this->city;
//    }
//
//    public function setCity(String $city): void
//    {
//        $this->city = $city;
//    }
//
//    public function getCountry(): String
//    {
//        return $this->country;
//    }
//
//    public function setCountry(String $country): void
//    {
//        $this->country = $country;
//    }
    // INITIAL )

    // SOLUTION (
    public function getStreetName(): ?String
{
    return $this->streetName;
}

    public function setStreetName(?String $streetName): Address
    {
        $this->streetName = $streetName;
        return $this;
    }

    public function getStreetNumber(): ?int
    {
        return $this->streetNumber;
    }

    public function setStreetNumber(?int $streetNumber): Address
    {
        $this->streetNumber = $streetNumber;
        return $this;
    }

    public function getCity(): ?String
    {
        return $this->city;
    }

    public function setCity(?String $city): Address
    {
        $this->city = $city;
        return $this;
    }

    public function getCountry(): ?String
    {
        return $this->country;
    }

    public function setCountry(?String $country): Address
    {
        $this->country = $country;
        return $this;
    }
    // SOLUTION )


}